jenkins/master_ctl.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python
+
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import getpass
+import subprocess
+import sys
+import time
+
+
+def Gcloud(project, *args, **kwargs):
+    cmd = ('gcloud', '--project=%s' % project, 'compute') + args
+    print >>sys.stderr, 'RUN:', ' '.join(cmd)
+    if not kwargs.get('output'):
+        subprocess.check_call(cmd)
+        return ''
+    return subprocess.check_output(cmd)
+
+
+def SnapshotDisks(project, zone, *disks):
+  ymd = time.strftime('%Y%m%d', time.localtime())
+  snapshots = ['%s-%s' % (d, ymd) for d in disks]
+  Gcloud(
+      project,
+      'disks',
+      'snapshot',
+      '--zone=%s' % zone,
+      '--snapshot-names=%s' % ','.join(snapshots),
+      *disks)
+
+
+def Disks(instance):
+    """Return a {name: size_gb} map."""
+    return {
+        instance: 500,
+        '%s-data' % instance: 1000,
+        '%s-docker' % instance: 1000,  # Only 200 in pr
+    }
+
+
+def Address(project, zone, instance):
+    """Return the reserved ip address of the instance."""
+    return Gcloud(
+        project,
+        'addresses',
+        'describe',
+        '%s-ip' % instance,
+        '--region=%s' % Region(zone),
+        '--format=value(address)',
+    )
+
+
+
+def Region(zone):
+    """Converts us-central1-f into us-central1."""
+    return '-'.join(zone.split('-')[:2])
+
+
+def Snapshot(project, zone, instance):
+    """Snapshot all the disks for this instance."""
+    SnapshotDisks(project, zone, *Disks(instance))
+
+
+def Delete(project, zone, instance):
```

Comment:
No, it doesn't delete the snapshots.

Any data that on the disks that isn't in a snapshot will be lost.